classes/db/SchemaDAO.inc.php

```diff
@@ -44,16 +44,20 @@ abstract class SchemaDAO extends DAO {
 	 * @return DataObject
 	 */
 	public function getById($objectId) {
+		// Try the cache first.
+		$pool = $this->_getCachePool();
+		$item = $pool->getItem($this->_getCacheId($objectId));
+		if (!$item->isMiss()) return $item->get();
+
+		// In case of cache miss, go to the database.
 		$result = $this->retrieve(
 			'SELECT * FROM ' . $this->tableName . ' WHERE ' . $this->primaryKeyColumn . ' = ?',
 			(int) $objectId
 		);
 
-		$returner = null;
-		if ($result->RecordCount() != 0) {
-			$returner = $this->_fromRow($result->GetRowAssoc(false));
-		}
+		$returner = $result->RecordCount() ? $this->_fromRow($result->GetRowAssoc(false)) : null;
 		$result->Close();
+		$pool->save($item->set($returner));
```

Comment:
@asmecher I'm wondering if this will cause a problem in cases where `_fromRow()` fetches dependent entities. So let's say I've got a `Submission` and it's in cache. Then I edit one of it's `Publication` objects. I've cleared the cache on the `Publication` object but not the `Submission`. When I go to get the `Submission` again it's going to return an object with nested `Publication`s, one of which will be out of date.

Reply:
Yes, I'm still working out details here... Off the top of my head, I'd like to exclude dependent entities from the serialized object and have them fetched separately (from the cache if possible) if we're going to use "deep caching" throughout the system. Another option (which is what we used to do) is to enable persistent caching only for high-volume scenarios like the reader front-end where mid-request updates don't happen.